src/Core/Services/GraphQLSchemaCreator.cs

```diff
@@ -227,12 +233,21 @@ private DocumentNode GenerateSqlGraphQLObjects(RuntimeEntities entities, Diction
                 }
             }
 
+            // For all the fields in the object which hold a foreign key reference to any referenced entity, add a foriegn key directive.
+            AddReferencingFieldDirective(entities, objectTypes);
```

Comment:
Fields from linking object and hence in linking object definition will always be referencing fields. The referencing field directive is added so that providing value for such fields becomes optional in insertion - because the values of such fields can come via insertion in referenced entity. But for linking entities, there is no question of insertion in the linking entity directly - unlike for an entity exposed in config. And hence we don't need to add the directive to referencing fields in linking entity.